web/src/components/profileEdit/QuotaCard.tsx

```diff
@@ -16,82 +16,112 @@
+  ProjectNamespaceResourceQuotaSize,
+} from '../../types';
+import { BaseIcon } from '../common/UI/Icon';
 
 interface IQuotaCardProps {
   quotaDetails: QuotaDetails;
+  href: string;
 }
 
 export interface QuotaDetails {
   licensePlate?: string;
-  quotaSize?: ProjectResourceQuotaSize;
-  quotaOptions?: {
-    quotaCpuSize: QuotaSize[];
-    quotaMemorySize: QuotaSize[];
-    quotaStorageSize: QuotaSize[];
-    quotaSnapshotSize: QuotaSize[];
+  quotaSize: ProjectSetResourceQuotaSize;
+  quotaOptions: {
+    dev: NamespaceQuotaOption;
+    test: NamespaceQuotaOption;
+    tools: NamespaceQuotaOption;
+    prod: NamespaceQuotaOption;
   };
 }
+
 interface SpecTextInterface {
   CPU: string;
   Memory: string;
   [key: string]: string;
 }
 
+export const NAMESPACE_DEFAULT_QUOTA: ProjectNamespaceResourceQuotaSize = {
+  quotaCpuSize: '',
+  quotaMemorySize: '',
+  quotaStorageSize: '',
+  quotaSnapshotSize: '',
+};
+
 const QuotaCard: React.FC<IQuotaCardProps> = (props) => {
   const {
     quotaDetails: {
       licensePlate = '',
       quotaSize = {
-        quotaCpuSize: '',
-        quotaMemorySize: '',
-        quotaStorageSize: '',
-        quotaSnapshotSize: '',
+        dev: NAMESPACE_DEFAULT_QUOTA,
+        test: NAMESPACE_DEFAULT_QUOTA,
+        tools: NAMESPACE_DEFAULT_QUOTA,
+        prod: NAMESPACE_DEFAULT_QUOTA,
       },
     },
+    href,
   } = props;
-
   const namespaceTexts = [
     ['Production', 'prod'],
     ['Test', 'test'],
     ['Development', 'dev'],
     ['Tools', 'tools'],
   ];
-  const specTexts: SpecTextInterface = {
-    CPU: quotaSize?.quotaCpuSize,
-    Memory: quotaSize?.quotaMemorySize,
-    Storage: quotaSize?.quotaStorageSize,
-    Snapshot: quotaSize?.quotaSnapshotSize,
-  };
 
   return (
     <Flex flexWrap="wrap">
-      {namespaceTexts.map((namespaceText: string[], index0: number) => (
-        <Aux key={index0}>
-          <Box width={1 / 2} px={2} mt={3}>
-            <Text as="h3">{namespaceText[0]} Namespace</Text>
-            <Text as="p" color={theme.colors.grey} fontSize={[1, 2, 2]} mt={1}>
-              {`${licensePlate}-${namespaceText[1]} namespace`}
-            </Text>
-          </Box>
-          <Box width={1 / 2} px={2} mt={3}>
-            <Text as="p" color={theme.colors.grey} fontSize={[2, 3, 3]} mt={1}>
-              CPU:{specTexts.CPU}
-            </Text>
-            <Text as="p" color={theme.colors.grey} fontSize={[2, 3, 3]} mt={1}>
-              Memory:{specTexts.Memory}
-            </Text>
-            <Text as="p" color={theme.colors.grey} fontSize={[2, 3, 3]} mt={1}>
-              Storage:{specTexts.Storage}
-            </Text>
-            <Text as="p" color={theme.colors.grey} fontSize={[2, 3, 3]} mt={1}>
-              Snapshot:{specTexts.Snapshot}
-            </Text>
-          </Box>
-        </Aux>
-      ))}
+      {namespaceTexts.map((namespaceText: string[], index0: number) => {
+        const index = namespaceText[1] as keyof typeof quotaSize;
```

Comment:
what the heck is this?

Reply:
idk, I just follow that pattern lol. Refined